Data/SBV/Core/SizedFloats.hs

```diff
@@ -87,7 +93,18 @@ data FP = FP { fpExponentSize    :: Int
              , fpSignificandSize :: Int
              , fpValue           :: BigFloat
              }
-             deriving (Ord, Eq, G.Data)
+             deriving (Eq, G.Data)
+
+-- Manually implemented instance as GHC generated a non-IEEE 754 compliant instance.
+-- Note that we cannot pack the values in a tuple and then compare them as that will
+-- also give non-IEEE 754 compilant results.
+instance Ord FP where
+  FP eb0 sb0 v0 <  FP eb1 sb1 v1 =
+    eb0 < eb1 || (eb0 == eb1 && (sb0 < sb1 || (sb0 == sb1 && v0 < v1)))
+  FP eb0 sb0 v0 <= FP eb1 sb1 v1 =
+    eb0 < eb1 || (eb0 == eb1 && (sb0 < sb1 || (sb0 == sb1 && v0 <= v1)))
```

Comment:
I think we should we throw an error here if `(eb0, sb0) /= (eb1, sb1)`, and if they're equal it should simply compare `v0` and `v1`

My motivation is that if this gets called with a differing sized float, then that's really a type error. (i.e., it should never happen) Don't you think?

Reply:
Also, we should bump libBF dependency to require >= 0.6.8 I suppose.

Reply:
Yeah that's a good idea. I implemented it like this just because this may be the original semantics, but fixed. Let me update it to throw errors.

For the libBF dependency, its `(<)` and `(<=)` are good, so we are able to implement a correct `Ord` instance even if the dependency is buggy.